src/libcore/slice/mod.rs

```diff
@@ -53,6 +53,20 @@ mod sort;
 // Extension traits
 //
 
+/// Calculates the direction and split point of a one-sided range.
+///
+/// Helper for `take` and `take_mut`,
+#[inline]
+fn take_split_point(range: impl OneSidedRange<usize>) -> (bool, usize) {
+    match (range.start_bound(), range.end_bound()) {
+        (Bound::Unbounded, Bound::Excluded(i)) => (true, *i),
+        (Bound::Unbounded, Bound::Included(i)) => (true, *i + 1),
+        (Bound::Excluded(i), Bound::Unbounded) => (false, *i + 1),
+        (Bound::Included(i), Bound::Unbounded) => (false, *i),
+        bounds => panic!("`OneSidedRange` returned invalid bounds: {:?}", bounds),
```

Comment:
Printing the actual bounds will probably increase the size of this function quite a bit. Bad early micro-optimizations, I know, but I feel like `take` could be used a lot in hot loops, so it should be quite speedy. Inlining this formatting machinery doesn't sound too great to me. Also, I am not sure how much printing the bounds actually helps anyone. This is just an internal bug. I would probably just use `unreachable!` here, I think.

Reply:
Done, although possibly there's merit to making `OneSidedRange` an unsafe trait so we can use `unreachable_unchecked`? I have basically no opinion here-- it seems likely that it gets inlined such that this branch is removed entirely.